modules/cassandra/cassandra.go

```diff
@@ -66,15 +73,72 @@ func WithInitScripts(scripts ...string) testcontainers.CustomizeRequestOption {
 				FileMode:          0o755,
 			}
 			initScripts = append(initScripts, cf)
-
 			execs = append(execs, initScript{File: cf.ContainerFilePath})
 		}
-
 		req.Files = append(req.Files, initScripts...)
 		return testcontainers.WithAfterReadyCommand(execs...)(req)
 	}
 }
 
+// SSLOptions contains the configuration options for setting up SSL/TLS
+type SSLOptions struct {
+	KeystorePath      string
+	KeystorePassword  string
+	CertPath          string
+	RequireClientAuth bool
+}
+
+// WithSSL enables SSL/TLS support on the Cassandra container
+func WithSSL(sslOpts SSLOptions) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		// Add the secure port to the exposed ports
+		hasSecurePort := false
+		for _, p := range req.ExposedPorts {
+			if p == string(securePort) {
+				hasSecurePort = true
+				break
+			}
+		}
+		if !hasSecurePort {
+			req.ExposedPorts = append(req.ExposedPorts, string(securePort))
+		}
+
+		// Set SSL environment variables
+		if req.Env == nil {
+			req.Env = make(map[string]string)
+		}
+
+		// If keystore path is provided, copy it to the container
+		if sslOpts.KeystorePath != "" {
+			keystoreFile := testcontainers.ContainerFile{
+				HostFilePath:      sslOpts.KeystorePath,
+				ContainerFilePath: "/etc/cassandra/conf/keystore.jks",
+				FileMode:          0o644,
+			}
+			req.Files = append(req.Files, keystoreFile)
+		}
+
+		if sslOpts.CertPath != "" {
+			certFile := testcontainers.ContainerFile{
+				HostFilePath:      sslOpts.CertPath,
+				ContainerFilePath: "/etc/cassandra/conf/cassandra.crt",
+				FileMode:          0o644,
+			}
+			req.Files = append(req.Files, certFile)
+		}
+
+		// Mark that SSL is enabled for later use
+		req.Labels = mergeMap(req.Labels, map[string]string{"testcontainers.cassandra.ssl": "true"})
```

Comment:
question: I guess this is needed to notify the Run function about the SSL addition, right? I'd choose a different pattern for this, creating an `options` type as in the redis module. Then, you can simply generate the SSL certs on-the-fly as in that module.